server/py/services/api/initial_data.py

```diff
@@ -877,41 +877,104 @@ def _perform_version_8_data_migrations(
 def _perform_version_9_data_migrations(
     db: framework.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    _migrate_function_kind(db, db_session, chunk_size=500)
+    _migrate_function_kind(db, db_session)
+    _migrate_artifact_producer_uri(db, db_session)
 
 
 def _migrate_function_kind(
```

Comment:
Although this is a part of a "migration", I still prefer `_ensure_function_kind` because we're not really migrating the kind from anywhere. Better be explicit on what we are actually doing here